runtime/acala/src/lib.rs

```diff
@@ -1429,6 +1442,40 @@ impl TakeRevenue for ToTreasury {
 	}
 }
 
+pub struct AcalaDropAssets<X, T>(PhantomData<(X, T)>);
+impl<X, T> DropAssets for AcalaDropAssets<X, T>
+where
+	X: DropAssets,
+	T: TakeRevenue,
+{
+	fn drop_assets(origin: &MultiLocation, assets: Assets) -> Weight {
```

Comment:
this is not a trivial method, so best to abstract it into the common runtime to avoid repeat this 3 times